drizzlepac/hapsequencer.py

```diff
@@ -1411,3 +1419,68 @@ def _verify_sci_hdrname(filename):
     if closefits:
         fhdu.close()
         del fhdu
+
+# ------------------------------------------------------------------------------
+
+
+def reset_keywords_apriori(filename):
+    """
+    Check the SCI extension(s) of the output FLT/FLC and DRZ/DRC files.  If the active
+    WCS solution is 'a priori', delete the following keywords if they are associated
+    with the active WCS as they are residue from a previous 'a posteriori' solution:
+    NMATCHES, RMS_RA/RMS_DEC, FITGEOM, and CRDER1/CRDER2. Ensure the WCSTYPE is based
+    upon the active WCSNAME to clarify any confusion.
+
+    Parameters
+    ----------
+    filename : str
+        Name of file to evaluate
+
+    Returns
+    -------
+    None
+    """
+    keys_to_delete = ['NMATCHES', 'RMS_RA', 'RMS_DEC', 'FITGEOM', 'CRDER1', 'CRDER2']
+
+    # Determine the number of SCI extensions
+    num_sci, extname = util.count_sci_extensions(filename)
+    extname_list = [(extname, x + 1) for x in range(num_sci)]
+
+    # Open the file in anticipation of updating 'a priori' solution associated keywords
+    hdu = fits.open(filename, mode='update')
+
+    # Loop over the SCI extensions in the file
+    for ext in extname_list:
+        active_wcsname = hdu[ext].header['WCSNAME']
+
+        # An 'a priori' solution is of the form <Starting WCS>-<Astrometric Catalog>
+        # Catalogs: GSC240, HSC30, GAIAD1, GAIADR2, and GAIAeDR3
+        wcsname_list = active_wcsname.split('-')
+
+        # If the active WCS solution is 'a priori', delete the keywords in the keys_to_delete
+        # list from the header which are remaining from a previous 'a posteriori' solution
+        is_apriori = len(wcsname_list) == 2 and wcsname_list[1].upper().startswith(('GSC', 'HSC', 'GAIA'))
+        if is_apriori:
+            try:
+                # Make sure the WCSTYPE is consistent with the active 'a priori' WCSNAME
+                # This keyword is not required for the WCS, but it provides an explanation
+                # in a succinct way.  If WCSTYPE does not exist, create it.
+                hdu[ext].header['WCSTYPE'] = updatehdr.interpret_wcsname_type(active_wcsname)
+
+            # Log a problem, but keep going as this keyword is a convenience.
+            except KeyError:
+                log.warning('Could not modify/add the WCSTYPE keyword for the active WCS solution.')
+
+            # Loop over the keys_to_delete list
+            for key in keys_to_delete:
+                try:
+                    if key in hdu[ext].header:
+                        del hdu[ext].header[key]
+
+                # Log a problem, but keep going as this is just a bit of cleanup
+                except KeyError:
+                    log.warning(f'Could not delete the {key} keyword from the active WCS solution.')
```

Comment:
There is no need for `try-except` here since there cannot be any exceptions since you are checking that the keyword exists in the header before deleting it. Please remove the try block.
```suggestion
            for key in keys_to_delete:
                if key in hdu[ext].header:
                    del hdu[ext].header[key]
```